configuration/types.go

```diff
@@ -396,6 +398,20 @@ type Configuration struct {
 	// but can use 10s of GBs of RAM, even with pruning enabled.
 	MemoryLimitDisabled bool `json:"memory_limit_disabled"`
 
+	// MaxTableSize default is 256 MB. The larger
+	// this value is, the larger database transactions
+	// storage can handle (~15% of the max table size
+	// == max commit size). It is only applied if MemoryLimitDisabled is true.
+	MaxTableSize int64 `json:"max_table_size,omitempty"`
```

Comment:
let's do these as `*int64`...we are going to do a migration to this so that we don't infer 0 as the default